R/zchunk_L210.water.demand.primary.R

```diff
@@ -34,45 +35,60 @@ module_water_L210.water.demand.primary_DISABLED <- function(command, ...) {
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are `merge` calls in this code. Be careful!
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # ===================================================
+
+
+      A21.globaltech_coef %>%
+        select(supplysector,subsector,technology) -> A21.tmp1
+       A22.globaltech_coef %>% filter( grepl('nuclear', supplysector)) %>%
+         select(supplysector,subsector,technology) -> A22.tmp
+       bind_rows(A21.tmp1,A22.tmp) -> A21.A22.tmp
+       left_join(L110.water_demand_primary_R_S_W_m3_GJ, A21.A22.tmp, by="supplysector") -> L210.TechCoef.v1
+
+
+      # A22.globaltech_coef %>%
+      #   filter(grepl("nuclear", supplysector)) %>%
+      #   select(supplysector, subsector, technology) %>%
+      #   bind_rows(select(A21.globaltech_coef, supplysector, subsector, technology)) %>%
+      #   left_join(L110.water_demand_primary_R_S_W_m3_GJ, by = "supplysector") ->
+      #   L210.TechCoef.v1
+
+      # Avoid double acounting unconventional oil in the regional oil sector as it is already
+      # accounted for in unconventional oil production.
+
+      L210.TechCoef.v1  %>%
+        filter(!(supplysector %in% "regional oil" & subsector %in% "unconventional oil")) %>%
+        mutate(water_sector = "Mining") %>%
+        mutate(minicam.energy.input = set_water_input_name(water_sector, water_type, A03.sector)) %>%
+        # Add in GCAM region names
+        left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+        mutate(market.name = region) -> L210.TechCoef.v2
+
+      # Set the coef for all years,and this is accomplished by:
+      # 1) repeat L210.TechCoef.v2 (672 rows) for 22 times (i.e., length(MODEL_YEARS)) and then bind rows;
+      # 2) add a column of "year", which is composed of 672 times of each element of MODEL_YEARS,
+      #    from MODEL_YEARS[1] to MODEL_YEARS[2],....., till the last element of MODEL_YEARS.
+      # 3) select 8 columns that is defined in "names_TechCoef" from the above tibble
+      L210.orig_num_rows <- nrow(L210.TechCoef.v2)
+      names_TechCoef <- c("region", "supplysector", "subsector", "technology","year","minicam.energy.input", "coefficient", "market.name")
+      L210.TechCoef.v2[rep(1:L210.orig_num_rows, times=length(MODEL_YEARS)),] %>%
+        mutate(year = MODEL_YEARS[sort(rep(1:length(MODEL_YEARS), times=L210.orig_num_rows))]) %>%
+        select(one_of(names_TechCoef)) -> results
+    }
```

Comment:
@bpbond Sorry for the mistake, I removed the `}`, but now I cannot commit the changes, it shows `commit failed`, is it because you pushed "Merge remote-tracking branch 'refs/remotes/origin/master' into Yaling-L210.water.demand.primary"?